### ISSUE TITLE
Use of union or other non-runtime types in NewType should generate error

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1397]
```python
A = NewType("A", Union[int, str])
B = NewType("B", Literal[1])
```

These should generate errors but do not.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.104, which I just published. It will also be included in the next release of pylance.
